nasa/harmony
# HARMONY-1848: Add additional validation for `bucketPath` parameter

NOTE: This is a breaking change as the allowed values are now more restricted

## Jira Issue ID
HARMONY-1848

## Description
Validate the `bucketPath` parameter to prevent people from sending malicious input like scripts

## Local Test Steps
1. Run this branch locally 
2. Try to send invalid characters in the`bucketPath` parameter when calling the bucket policy endpoint
```
http://localhost:3000/staging-bucket-policy?bucketPath=%3Cscript%3Efoo()%3C/script%3E
```
3. Check the bucket policy endpoint documentation in the harmony online docs and verify that it links to guidelines for bucket names and valid path characters
```
http://localhost:3000/docs#user-owned-buckets-for-harmony-output 
```

## PR Acceptance Checklist
